pkg/apis/application/v1alpha1/types_test.go

```diff
@@ -34,6 +34,14 @@ func TestAppProject_IsSourcePermitted(t *testing.T) {
 		projSources: []string{"https://github.com/argoproj/*.git"}, appSource: "https://github.com/argoproj1/test2.git", isPermitted: false,
 	}, {
 		projSources: []string{"https://github.com/argoproj/foo"}, appSource: "https://github.com/argoproj/foo1", isPermitted: false,
+	}, {
+		projSources: []string{"https://gitlab.com/group/*"}, appSource: "https://gitlab.com/group/repo/owner", isPermitted: true,
```

Comment:
I would have expected that this pattern would not match, because it uses a single asterisk.

Reply:
@jannfis have added a `/` separator in sources.